### PR TITLE
Pokemon List cleanup and Mega data

### PR DESCRIPTION
**Pull request by Qwertypop04**
Moves all Pokemon IDs to n.0x where some were previously n.x, and moves around the ordering of some alternate forms
Moves all Mega forms to be normal alt forms, adds their Mega Evolutions, and Mega Stones. This does not make any more Megas available, just makes it easier to make more available later.

### CODE REVIEW

File: src/scripts/Update.ts [hunk 1]
```diff
@@ -1420,6 +1420,207 @@ class Update implements Saveable {
...
+                [877.1, 877.01],
+                [888.1, 888.01],
+                [889.1, 889.01],
+                [892.1, 892.01],
+                [893.1, 893.01],
+                [898.1, 898.01],
+                [898.2, 898.02],
+            ];
+
+            consistentIDs.forEach(([oldID, newID]) => {
```
Comment (Aegyo) [inline src/scripts/Update.ts:1591]: This isn't going to work for the Alcremie ones, you are overwiting data of a different Alcremie, so it won't be there when you try to move that id later. Osherwott is working on a method of rotating ids around in #3340, I would wait for that before these Alcremie changes.

eg .53 overwrites into .54, and then later you try to move .54 into .51, but .54 now contains stats for the old .53

**Comment by osherwott** [comment]
Would it be possible for you to move Pikachu (World Cap) to after (Partner Cap), since that way it would more accurately reflect the main games' pokedex entries and pokeclicker's acquisition order? Everything else looks stellar! We'll try to get that update function out as soon as possible!

**Comment by Qwertypop04** [comment]
> Would it be possible for you to move Pikachu (World Cap) to after (Partner Cap), since that way it would more accurately reflect the main games' pokedex entries and pokeclicker's acquisition order? Everything else looks stellar! We'll try to get that update function out as soon as possible!

No. The way it is is intentional.

**Comment by Aegyo** [comment]
What's the reasoning?

**Comment by Qwertypop04** [comment]
> What's the reasoning?

It is essentially the "Galar Cap". So it is grouped with the rest of the region caps. Partner Cap is a seperate thing.

File: src/scripts/Update.ts [hunk 2]
```diff
@@ -1513,6 +1518,253 @@ class Update implements Saveable {
...
+                }
+                // Update our statistics
+                saveData.statistics.pokemonEncountered[newID] = saveData.statistics.pokemonEncountered[oldID] || 0;
+                saveData.statistics.pokemonDefeated[newID] = saveData.statistics.pokemonDefeated[oldID] || 0;
+                saveData.statistics.pokemonCaptured[newID] = saveData.statistics.pokemonCaptured[oldID] || 0;
+                saveData.statistics.pokemonHatched[newID] = saveData.statistics.pokemonHatched[oldID] || 0;
+                saveData.statistics.shinyPokemonEncountered[newID] = saveData.statistics.shinyPokemonEncountered[oldID] || 0;
+                saveData.statistics.shinyPokemonDefeated[newID] = saveData.statistics.shinyPokemonDefeated[oldID] || 0;
+                saveData.statistics.shinyPokemonCaptured[newID] = saveData.statistics.shinyPokemonCaptured[oldID] || 0;
+                saveData.statistics.shinyPokemonHatched[newID] = saveData.statistics.shinyPokemonHatched[oldID] || 0;
```
Comment (Aegyo) [inline src/scripts/Update.ts:1757]: Needs the male/female variants of all of these too

Comment (Qwertypop04) [reply src/scripts/Update.ts:1757]: Ah, of course. I believe that should be all of them?

**Review by Jaaslet — APPROVED** [review]
I have fixed the problems i found. There are no more missingNOs, and i tested a couple of mons to see if the stats are transferred correct, and i found no problems.
Thanks for the PR

**Pull request merged**